src/components/vm/overview/watchdog.tsx

```diff
@@ -97,39 +97,46 @@ export const WatchdogModal = ({
 
     const Dialogs = useDialogs();
 
-    const setWatchdogColdplug = () => {
-        return domainSetWatchdog({
-            connectionName: vm.connectionName,
-            vmName: vm.name,
-            action: watchdogAction,
-            defineOffline: true,
-            hotplug: false,
-            isWatchdogAttached,
-        })
-                .then(Dialogs.close, exc => setDialogError({ text: _("Failed to configure watchdog"), detail: exc.message, variant: "danger" }));
+    async function setWatchdog(vm: VM, hotplug: boolean) {
+        if (isWatchdogAttached) {
+            await virtXmlEdit(vm, "watchdog", 1, { action: watchdogAction });
+        } else {
+            await virtXmlAdd(
+                vm,
+                "watchdog",
+                { action: watchdogAction },
+                { update: hotplug, "no-define": hotplug },
+            );
+        }
+    }
+
+    const setWatchdogColdplug = async () => {
+        try {
+            await setWatchdog(vm, false);
+            Dialogs.close();
+        } catch (exc) {
+            setDialogError({ text: _("Failed to configure watchdog"), detail: String(exc), variant: "danger" });
+        }
     };
 
-    const setWatchdogHotplug = () => {
-        return domainSetWatchdog({
-            connectionName: vm.connectionName,
-            vmName: vm.name,
-            action: watchdogAction,
-            defineOffline: false,
-            hotplug: true,
-            isWatchdogAttached,
-        })
-                .then(vm.persistent ? setWatchdogColdplug : Dialogs.close)
-                .catch(exc => {
-                    if (vm.persistent)
-                        setOfferColdplug(true);
+    const setWatchdogHotplug = async () => {
+        try {
+            await setWatchdog(vm, true);
+            if (vm.persistent)
+                await setWatchdogColdplug();
+            else
+                Dialogs.close();
+        } catch (exc) {
+            if (vm.persistent)
+                setOfferColdplug(true);
 
-                    setDialogError({
-                        text: _("Could not dynamically add watchdog"),
-                        detail: _("Adding a watchdog will require a reboot to take effect."),
-                        expandableDetail: exc.message,
-                        variant: vm.persistent ? "warning" : "danger"
-                    });
-                });
+            setDialogError({
+                text: _("Could not dynamically add watchdog"),
+                detail: _("Adding a watchdog will require a reboot to take effect."),
+                expandableDetail: String(exc),
+                variant: vm.persistent ? "warning" : "danger"
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2334-68be91e9-20250929-092324-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)